packages/utils/src/lib/reports/formatting.ts

```diff
@@ -58,3 +66,58 @@ export function metaDescription(
   }
   return '';
 }
+
+/**
+ * Link to local source for IDE
+ * @param source
+ * @param reportLocation
+ *
+ * @example
+ * linkToLocalSourceInIde({ file: '/src/index.ts'}, {outputDir: '/.code-pushup'}) // [/src/index.ts](../src/index.ts)
+ */
+export function linkToLocalSourceForIde(
+  source: SourceFileLocation,
+  options?: Pick<MdReportOptions, 'outputDir'>,
+): InlineText {
+  const { file, position } = source;
+  const { outputDir } = options ?? {};
+  const unixPath = toUnixPath(file);
+
+  // NOT linkable
+  if (!outputDir) {
+    return md.code(unixPath);
+  }
+
+  const relativePath = pathPosix.relative(outputDir, unixPath);
+  return md.link(formatFilePosition(relativePath, position), md.code(unixPath));
+}
+
+export function formatSourceLine(source: SourceFileLocation): string {
+  if (!source.position) {
+    return '';
+  }
+  const { startLine, endLine } = source.position;
+  return endLine
+    ? startLine === endLine
+      ? `${startLine}`
+      : `${startLine}-${endLine}`
+    : `${startLine}`;
+}
+
+export function formatFilePosition(
+  file: string,
+  position?: SourceFileLocation['position'],
+): string {
+  if (!position) {
+    return file;
+  }
+  const ide = ideEnvironment();
+  const { startLine, endLine } = position;
+  return ide === 'vscode'
+    ? `${file}#L${startLine}`
+    : ide === 'github'
+    ? endLine
+      ? `${file}#L${startLine}-L${endLine}`
+      : `${file}#L${startLine}`
```

Comment:
I don't understand how this is intended to work for GitHub Actions. The `report.md` can be downloaded as an artifact, but we don't have any control over where they'll open it, so the relative links are likely to be broken.

It would make more sense to link to remote URL in this case. Since we already know we're running in a GitHub Actions environment, we can use its [built-in environment variables](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables) to reliably construct the correct GitHub URL. Something like:

```ts
`${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}/blob/${process.env.GITHUB_SHA}/${issue.source.file}`
````